traits/trait_types.py

```diff
@@ -1430,6 +1431,9 @@ def validate(self, object, name, value):
 
             Note: The 'fast validator' version performs this check in C.
         """
+        if isinstance(value, PathLike):
+            value = fspath(value)
```

Comment:
If you give `os.fspath` a string it will return it unmodified: https://docs.python.org/3/library/os.html#os.fspath

So it is better to  just call it in a `try: ... except TypeError: ...`, I think.  if the exception happens then it doesn't validate, otherwise you have a string which is going to hold a filesystem path.